src/Parser/Shaman/Elemental/Modules/ShamanCore/EarthShock.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import ItemLink from 'common/ItemLink';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { formatNumber } from 'common/format';
+
+class EarthShock extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+    combatants: Combatants,
+  };
+
+  maelstromSpendOutAscendence = [];
+  maelstromSpendInAscendence = [];
+  latershock = false;
+  gambling = false;
+
+  on_initialized() {
+    this.gambling = (this.combatants.selected.hasHands(ITEMS.SMOLDERING_HEART.id) && this.combatants.selected.hasFeet(ITEMS.THE_DECEIVERS_BLOOD_PACT.id));
+    this.pseudogambling = (this.combatants.selected.hasShoulder(ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id) && this.combatants.selected.hasFeet(ITEMS.THE_DECEIVERS_BLOOD_PACT.id));
+  }
+
+  get avgMaelstromSpendOutOfAsc() {
+    let spend = this.maelstromSpendOutAscendence.reduce((prev, cur) => {
+      return prev + cur;
+    }, 0);
+    spend /= this.maelstromSpendOutAscendence.length || 1;
+    return spend;
+  }
+  get avgMaelstromSpendInAsc() {
+    let spend = this.maelstromSpendInAscendence.reduce((prev, cur) => {
+      return prev + cur;
+    },0);
+    spend /= this.maelstromSpendInAscendence.length||1;
+    return spend;
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid === SPELLS.EARTH_SHOCK.id && !this.latershock) {
+      this.latershock = true;
+      if (event.classResources[0].type === RESOURCE_TYPES.MAELSTROM.id) {
+        if (this.combatants.selected.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id)) {
+          this.maelstromSpendInAscendence.push(event.classResources[0].amount);
+        }
+        else {
+          this.maelstromSpendOutAscendence.push(event.classResources[0].amount);
+        }
+      }
+    }
+    if (event.classResources[0].amount < 40 && event.classResources[0].type === RESOURCE_TYPES.MAELSTROM.id) {
+      this.latershock = false;
+    }
+  }
+
+  suggestions(when) {
```

Comment:
(forgot to submit this comment)

This section needs overhaul in my opinion. Please use the `actual` and `recommended` variables when you can. But more importantly, I would advise you to use the "suggestion objects" some of us are using.

Basically means you'll make 3 getters for each `gambling`, `pseudogambling` and "default", which return an object representing the suggestion thresholds and values, and pass it into the `when` function which takes care of the rest. It's easier to maintain if you decide to change the values, and as a bonus - it's also much more convenient to use when you get to implementing a Checklist in the future (see any implemented one for example).

An example for the "default" would be:
```jsx
get defaultSuggestionThresholds() {
    return {
      actual: this.avgMaelstromSpendOutOfAsc,
      isLessThan: {
        minor: 111,   // these numbers don't make a lot of sense either, please take a look at them
        average: 5,
        major: 15,
      },
      style: 'number',
    };
  }

suggestions(when) {
   .....
   else {
       when(this.defaultSuggestionThresholds)
          .addSuggestion((suggest, actual, recommended) => {
            return suggest(<span>Try to cast <SpellLink id={SPELLS.EARTH_SHOCK.id} /> at 111 Maelstrom or more.</span>)
            .icon(SPELLS.EARTH_SHOCK.icon)
            .actual(`Average Maelstrom spent: ${formatNumber(actual)}`)
            .recommended(`> ${recommended} is recommended`);
       });
   }
}
```

Reply:
Ah i thought the average and minor values were deltas from the suggested value, that makes things a little different then ye.

Reply:
Don't know if that would help out a lot tho because the wording changes in every case, so i would need to  make that dynamic too and also it get's weird with recommended being a range.

Reply:
The average and minor values were always a concrete number (that's why people wrote it like `regular(recommended + 5)`). Also, I think you misunderstood me - the object is there just to abstract the numbers away, you will still have 3 different suggestions with different texts, just the numbers are elsewhere.

Reply:
okay that makes sense then thx